### PR TITLE
Made fake implementations partial

### PR DESCRIPTION
**Pull request by timbussmann**
Make them partial following @danielmarbach suggestion to allow easier extension if necessary in the testing repository without needing to touch the source files from the source only package

Connects to Particular/NServiceBus.Testing#29

### CODE REVIEW

**Comment by danielmarbach** [comment]
Tests failing

**Comment by timbussmann** [comment]
weird, seems to be random failing encryption test. Triggered a new build. Will have a look at that test.

**Comment by andreasohlund** [comment]
Yep the encryption test is flaky, pretty sure @MarcinHoppe raised an issue for that

**Comment by MarcinHoppe** [comment]
#3365 ?

**Comment by andreasohlund** [comment]
@timbussmann was it the same failure as in #3365?

**Comment by timbussmann** [comment]
@MarcinHoppe @andreasohlund not exactly the same test, but same tests class yes. See build: https://builds.particular.net/viewLog.html?buildId=155920&tab=buildResultsDiv&buildTypeId=NServiceBusCore_Build

File: src/NServiceBus.Testing.Fakes/OutgoingMessageExtensions.cs [hunk 1]
```diff
@@ -40,13 +40,6 @@ public static IEnumerable<SentMessage<TMessage>> Containing<TMessage>(this IEnum
                 .Select(x => new SentMessage<TMessage>((TMessage) x.Message, x.Options));
         }
 
-        internal static IEnumerable<TimeoutMessage<TMessage>> Containing<TMessage>(this IEnumerable<TimeoutMessage<object>> timeoutMessages)
```
Comment (danielmarbach) [inline src/NServiceBus.Testing.Fakes/OutgoingMessageExtensions.cs]: Forgive my ignorance, but why was this removed?

File: src/NServiceBus.Testing.Fakes/TimeoutMessage.cs [hunk 2]
```diff
@@ -1,25 +0,0 @@
-﻿namespace NServiceBus.Testing
-{
-    using System;
-
-    /// <summary>
-    /// Represents a message that is timed out, if not processed within a given timespan. Contains the message itself and its
-    /// associated options.
-    /// </summary>
-    /// <typeparam name="TMessage">The message type</typeparam>
-    public class TimeoutMessage<TMessage> : OutgoingMessage<TMessage, SendOptions>
```
Comment (danielmarbach) [inline src/NServiceBus.Testing.Fakes/TimeoutMessage.cs]: Forgive my ignorance, but why was this removed?

**Comment by danielmarbach** [comment]
Two questions. Apart from that all good

File: src/NServiceBus.Testing.Fakes/TimeoutMessage.cs (hunk 2 above)
Comment (timbussmann) [inline src/NServiceBus.Testing.Fakes/TimeoutMessage.cs]: this is only used by the NServiceBus.Testing `TestingContext` class and should live in that repository.

File: src/NServiceBus.Testing.Fakes/OutgoingMessageExtensions.cs (hunk 1 above)
Comment (timbussmann) [inline src/NServiceBus.Testing.Fakes/OutgoingMessageExtensions.cs]: TimeoutMessage is only relevant for NServiceBus.Testing. Since this is internal, it should be declared in the NServiceBus.Testing repo.

**Comment by MarcinHoppe** [comment]
@timbussmann @andreasohlund Added this link to failed tests to #3365.
